File: PostApp/app/src/main/java/hootor/com/postapp/view/MainActivity.java
package hootor.com.postapp.view;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import hootor.com.postapp.R;
import hootor.com.postapp.model.User;
import hootor.com.postapp.service.PostAppService;
import hootor.com.postapp.service.RetrofitInstance;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private EditText userEmail;
    private EditText passWord;
    private Button submitButton;
    private TextView resultTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        userEmail = findViewById(R.id.et_email);
        passWord = findViewById(R.id.et_password);
        submitButton = findViewById(R.id.btn_submit);
        resultTextView = findViewById(R.id.tv_result);

        submitButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                postData();
            }
        });
    }

    private void postData() {

        User user = new User();
        user.setUserEmail(userEmail.getText().toString().trim());
        user.setPassWord(passWord.getText().toString().trim());

        PostAppService postAppService = RetrofitInstance.getService();
        Call<User> call = postAppService.getResult(user);

        userEmail.setText("");
        passWord.setText("");

        call.enqueue(new Callback<User>() {
            @Override
            public void onResponse(Call<User> call, Response<User> response) {

                User returnedUser = response.body();
                resultTextView.setText("Id is: " + returnedUser.getId());
            }

            @Override
            public void onFailure(Call<User> call, Throwable t) {

            }
        });
    }
}

File: RetrofitDemo/app/src/main/java/com/android/hootr/retrofitdemo/MainActivity.kt
package com.android.hootr.retrofitdemo

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.android.hootr.retrofitdemo.adapters.CountryAdapter
import com.android.hootr.retrofitdemo.model.CountryInfo
import com.android.hootr.retrofitdemo.model.Result
import com.android.hootr.retrofitdemo.service.RetrofitInstance
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    var resultArrayLiist: ArrayList<Result> = arrayListOf<Result>()
    val countryApapter = CountryAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView.layoutManager = LinearLayoutManager(this)

        recyclerView.adapter = countryApapter

        getCountries()
    }

    private fun getCountries() {

        val countryService = RetrofitInstance.getService()
        val call = countryService.getResults()

        call.enqueue(object : Callback<CountryInfo> {
            override fun onResponse(call: Call<CountryInfo>, response: Response<CountryInfo>) {

                val countryInfo = response.body()
                countryInfo?.let { it ->
                    resultArrayLiist = it.restResponse.result
                    countryApapter.resultArrayList = resultArrayLiist
                }

            }

            override fun onFailure(call: Call<CountryInfo>, t: Throwable) {
                Log.i("happy", t.toString())

            }
        })

    }
}
